### DISCUSSION TITLE
Default cache cleaner mixin not working

### DISCUSSION THREAD

**Discussion by Kathir-1995** [moleculerjs/moleculer#1281] [Q&A]
<html>
<body>

module.exports = function(serviceNames) {    const events = {};    serviceNames.forEach(name => {        events[`cache.clean.${name}`] = function() {            if (this.broker.cacher) {                this.logger.debug(`Clear local '${this.name}' cache`);                this.broker.cacher.clean(`${this.name}.*`);            }        };    });    return {        events    };};
--

I have used the cache cleaner mixin in my service, After updating the data, still i am getting the cached data and the mixin was not properly cleared the cache. Anyone suggest solution for this issue?

</body>
</html>

**Comment by icebob** [comment]
cache cleaner just clear the cache entries if it receives the `cache.clean.<servicename>` event. Do you emit this event?

**Reply by Kathir-1995** [reply]
Yes i have used the cache cleaner mixin in my service as per the documentation
